src/Guzzle/SubjectRewritingMiddleware.php

```diff
@@ -0,0 +1,206 @@
+                    case 'application/vnd.elife.press-package-list+json; version=1':
+                    case 'application/vnd.elife.recommendations+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+
+                    case 'application/vnd.elife.article-related+json; version=1':
+                        $data = $this->updateItems($data);
+                        break;
+
+                    case 'application/vnd.elife.article-poa+json; version=1':
+                    case 'application/vnd.elife.article-poa+json; version=2':
+                    case 'application/vnd.elife.article-vor+json; version=1':
+                    case 'application/vnd.elife.article-vor+json; version=2':
+                    case 'application/vnd.elife.blog-article+json; version=2':
+                        $data = $this->updateItem($data);
+                        break;
+
+                    case 'application/vnd.elife.collection+json; version=1':
+                        $data = $this->updateItem($data);
+                        $data['content'] = $this->updateItems($data['content']);
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.cover-list+json; version=1':
+                    case 'application/vnd.elife.highlight-list+json; version=1':
+                        $data['items'] = array_map(function (array $cover) {
+                            $cover['item'] = $this->updateItem($cover['item']);
+
+                            return $cover;
+                        }, $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.podcast-episode+json; version=1':
+                        $data['chapters'] = array_map(function (array $chapter) {
+                            $chapter['content'] = $this->updateItems($chapter['content'] ?? []);
+
+                            return $chapter;
+                        }, $data['chapters']);
+                        break;
+
+                    case 'application/vnd.elife.person-list+json; version=1':
+                        $data['items'] = array_map([$this, 'updatePerson'], $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.person+json; version=1':
+                        $data = $this->updatePerson($data);
+                        break;
+
+                    case 'application/vnd.elife.press-package+json; version=3':
+                        $data = $this->updateItem($data);
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.search+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
```

Comment:
the level of nesting here would warrant a private method dedicated to search results

Reply:
Extracted the search subjects part.